Sources/AsyncAlgorithms/Deadline.swift

```diff
@@ -0,0 +1,60 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Async Algorithms open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+public struct TimeoutError<C: Clock>: Error {
```

Comment:
For consistency this likely should be a `DeadlineExceededError`, we don't speak in terms of timeouts; timeout is the convenience API

Reply:
Yea that might be a better name.